### PR TITLE
Change experiments table circle to radio button

### PR DESCRIPTION
**Pull request by mattseddon**
# 1/2 `main` <- this <- #2567 

Related to https://github.com/iterative/vscode-dvc/issues/2373

This PR updates the experiments table and replaces the existing circles with a radio button. In the follow-up PR I have tried to improve the tooltip displayed when hovering over the radio button. Although not perfect I think this change is an improvement and good enough to ship.

### Demo

https://user-images.githubusercontent.com/37993418/195230528-eb050f12-2089-461b-94aa-cbcd585b996d.mov

Previous attempts:

### Demo

https://user-images.githubusercontent.com/37993418/194790356-651c4cac-4cc8-478e-9410-8757c3860da5.mov

#### With the sub-row indicator moved

https://user-images.githubusercontent.com/37993418/194805128-7b0f5ab7-29a3-48f1-9b01-6400254cca63.mov

Left in draft and asked for reviews because I would like everyone to take a look at the idea.

I think we should move the eye + circle to the LHS with the other row action indicators and put any required whitespace between the circle and chevron. Like this:

image

### Demo of the above

https://user-images.githubusercontent.com/37993418/194820317-9e7f77b9-c8a1-4d75-8283-58f30d357517.mov

### CODE REVIEW

**Comment by maxagin** [comment]
What is going to be displayed in the exps sidebar section?

**Comment by mattseddon** [comment]
> What is going to be displayed in the exps sidebar section?

No change

**Comment by julieg18** [comment]
Adding the eye definitely makes things more clear for me! 

> I think we should move the eye + circle to the LHS with the other row action indicators and put any required whitespace between the circle and chevron. 

➕  on this idea!

**Review by sroy3 — APPROVED** [review]
Now that we are having more actions to the left, would it make sense to more the label of the column to something like this?
Screen Shot 2022-10-12 at 9 05 33 AM

Or even move the name to the right?
Screen Shot 2022-10-12 at 9 08 13 AM
(this would need to make that column be able to resize smaller than it possible at the moment)

File: webview/src/experiments/components/table/CellRowActions.tsx [hunk 1]
```diff
@@ -51,11 +59,17 @@ export const CellRowAction: React.FC<CellRowActionProps> = ({
   )
 }
 
+const getTooltipContent = (determiner: boolean, text: string): string =>
```
Comment (julieg18) [inline webview/src/experiments/components/table/CellRowActions.tsx:62]: Great idea for creating the text!

File: webview/src/experiments/components/table/CellRowActions.tsx [hunk 2]
```diff
@@ -86,10 +100,29 @@ export const CellRowActions: React.FC<CellRowActionsProps> = ({
...
+          </span>
+        </div>
+      ) : (
+        <CellRowAction
+          showSubRowStates={showSubRowStates}
+          subRowsAffected={plotSelections}
+          testId={'row-action-plot'}
+          tooltipContent={getTooltipContent(!!bulletColor, 'Plot')}
+        >
+          <span
```
[the remark this answers is not in the document]
Comment (julieg18) [reply webview/src/experiments/components/table/CellRowActions.tsx:119]: ➕

**Review by julieg18 — APPROVED** [review]
Great work, Matt! 

> Now that we are having more actions to the left, would it make sense to move the label of the column to something like this?

> Or even move the name to the right?

> (this would need to make that column be able to resize smaller than it possible at the moment)

Personally, I find the `Experiments` label being in the "center" to look like of buggy. As for the second option, I believe it is recommended that we keep text left-aligned instead of right-aligned for readability. No strong opinion on this though :)

**Comment by shcheklein** [comment]
One caveat (not critical for now) is checkpoints (and resume / continue logic) when line breaks between the checkpoints. In this case line gives a hint.

LGTM otherwise, let's proceed with this. I would say though that I don't think this is enough. I think we need more explicit things, buttons, etc to connect both webviews.

File: webview/src/experiments/components/table/styles.module.scss [hunk 3]
```diff
@@ -95,19 +95,26 @@ $workspace-row-edge-margin: $edge-padding - $cell-padding;
...
+  width: calc(var(--design-unit) * 4px);
   color: $fg-color;
-  display: inline-block;
-  width: 24px;
-  flex: 0 0 24px;
-  text-align: center;
+
+  align-items: center;
+  justify-content: center;
+  align-self: center;
```
[the remark this answers is not in the document]
Comment (mattseddon) [reply webview/src/experiments/components/table/styles.module.scss:111]: removed

File: webview/src/experiments/components/table/styles.module.scss [hunk 4]
```diff
@@ -95,19 +95,26 @@ $workspace-row-edge-margin: $edge-padding - $cell-padding;
...
 
 .bullet {
+  visibility: visible;
+  background: $checkbox-background;
+  border-radius: 999px;
+  border: calc(var(--border-width) * 1px) solid $checkbox-border;
+  box-sizing: border-box;
+  cursor: pointer;
+  height: calc(var(--design-unit) * 4px);
+  outline: none;
```
[the remark this answers is not in the document]
Comment (mattseddon) [reply webview/src/experiments/components/table/styles.module.scss:105]: changed to a button

File: webview/src/experiments/components/table/styles.module.scss [hunk 5]
```diff
@@ -95,19 +95,26 @@ $workspace-row-edge-margin: $edge-padding - $cell-padding;
 }
 
 .bullet {
+  visibility: visible;
+  background: $checkbox-background;
+  border-radius: 999px;
+  border: calc(var(--border-width) * 1px) solid $checkbox-border;
+  box-sizing: border-box;
+  cursor: pointer;
+  height: calc(var(--design-unit) * 4px);
```
[the remark this answers is not in the document]
Comment (mattseddon) [reply webview/src/experiments/components/table/styles.module.scss:104]: pulled out variable

**Comment by mattseddon** [comment]
> Now that we are having more actions to the left, would it make sense to move the label of the column to something like this? Screen Shot 2022-10-12 at 9 05 33 AM
> 
> Or even move the name to the right? Screen Shot 2022-10-12 at 9 08 13 AM (this would need to make that column be able to resize smaller than it possible at the moment)

@sroy3 this is good. I managed to move "Experiment" to the right but failed with the experiment names. Can you help me out with the code?

**Comment by mattseddon** [comment]
@julieg18 @sroy3 for moving experiment names to the right of the first cell please take a look and the storybook and last commit. LMK what you think, thanks.

**Comment by julieg18** [comment]
> The last one is probably my preferred option.
> WDYT? Am I overthinking this?

Last one is also my preferred option! The second one looks like of buggy to me with how the text jumps and second one shows what appears to be secondary text (gray and small) before the primary text (white and large).

File: webview/src/experiments/components/table/CellRowActions.tsx [hunk 6]
```diff
@@ -86,10 +100,29 @@ export const CellRowActions: React.FC<CellRowActionsProps> = ({
...
+        <CellRowAction
+          showSubRowStates={showSubRowStates}
+          subRowsAffected={plotSelections}
+          testId={'row-action-plot'}
+          tooltipContent={getTooltipContent(!!bulletColor, 'Plot')}
+        >
+          <button
+            className={styles.bullet}
+            style={{ color: bulletColor }}
+            {...clickAndEnterProps(toggleExperiment)}
```
[the remark this answers is not in the document]
Comment (mattseddon) [reply webview/src/experiments/components/table/CellRowActions.tsx]: This ended up being a button inside a button so I had to move the action into `<Indicator/>`

[the remark this answers is not in the document]
Comment (mattseddon) [reply webview/src/experiments/components/table/CellRowActions.tsx]: I will make the same change for the other row actions in a follow-up (the change is bigger than I anticipated).

**Comment by shcheklein** [comment]
Q: how will to look like with three commits? (my concern that we've completely flattened out the table - commit name, experiments (and checkpoints?) are all on the same level now? If that is the case, then we are loosing important information - hierarchy.

**Comment by mattseddon** [comment]
> Q: how will to look like with three commits? (my concern that we've completely flattened out the table - commit name, experiments (and checkpoints?) are all on the same level now? If that is the case, then we are loosing important information - hierarchy.

image

**Comment by shcheklein** [comment]
Hmm, I prefer it way better the way it was before. It was more informative (in an important way - there is child-parent relationship). We've removed lines and hierarchy. It means either we'll have to implement something like in studio (sections per branch) or we need to go back and introduce the hierarchy back.

**Comment by mattseddon** [comment]
> Hmm, I prefer it way better the way it was before.

Which part of the change?

**Comment by shcheklein** [comment]
Sorry, I meant the names. Buttons are fine and definitely an improvement!

**Comment by mattseddon** [comment]
> Sorry, I meant the names. Buttons are fine and definitely an improvement!

Ok, I will raise a PR to revert.

**Pull request merged**